Rubberduck.Parsing/Annotations/AttributeAnnotationBase.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections.Generic;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.Annotations
+{
+    public abstract class AttributeAnnotationBase : AnnotationBase, IAttributeAnnotation
+    {
+        protected AttributeAnnotationBase(AnnotationType annotationType, QualifiedSelection qualifiedSelection, VBAParser.AnnotationContext context, IReadOnlyList<string> attributeValues) 
```

Comment:
woud `params string[] values` be more practical?

Reply:
Definetly not. I have to pass this to this base class from the derived constructor. Moreover, this is exactly what we want to save on the annotation. In addition, the annotation constructors get called via reflection in the `AnnotationFactory`, passing in a list. Again, a parameter array would just complicate things.

Reply:
Ah, I missed the reflection bit :+1:

Reply:
@MDoerner hold on, I misplaced that comment. What I meant is that these:

     : base(AnnotationType.DefaultMember, qualifiedSelection, context, new List<string> { "0" })
     : base(AnnotationType.Description, qualifiedSelection, context, parameters?.Take(1).ToList())
     : base(AnnotationType.Enumerator, qualifiedSelection, context, new List<string> { "-4" })
     : base(AnnotationType.Exposed, qualifiedSelection, context, new List<string> { Tokens.True })

Could be written like this instead, if the base class took a `params` array for the `attributeValues` argument.

     : base(AnnotationType.DefaultMember, qualifiedSelection, context, "0")
     : base(AnnotationType.Description, qualifiedSelection, context, parameters?.First())
     : base(AnnotationType.Enumerator, qualifiedSelection, context, "-4")
     : base(AnnotationType.Exposed, qualifiedSelection, context, Tokens.True)

It doesn't look like the `AnnotationFactory` would be affected.

Reply:
Nevermind

    :base(AnnotationType.MemberAttribute, qualifiedSelection, context, parameters?.Skip(1).ToList())